apps/cowswap-frontend/src/modules/appData/hooks.ts

```diff
@@ -5,19 +5,19 @@ import { DEFAULT_APP_CODE, SAFE_APP_CODE } from '@cowprotocol/common-const'
 import { isInjectedWidget } from '@cowprotocol/common-utils'
 import { useIsSafeApp } from '@cowprotocol/wallet'
 
+import { useInjectedWidgetMetaData } from 'modules/injectedWidget'
+
 import { addAppDataToUploadQueueAtom, appDataHooksAtom, appDataInfoAtom } from './state/atoms'
 import { AppDataInfo } from './types'
 
-import { injectedWidgetMetaDataAtom } from '../injectedWidget/state/injectedWidgetMetaDataAtom'
-
 const APP_CODE = process.env.REACT_APP_APP_CODE
 
 export function useAppData(): AppDataInfo | null {
   return useAtomValue(appDataInfoAtom)
 }
 
 export function useAppCode(): string | null {
-  const injectedWidgetMetaData = useAtomValue(injectedWidgetMetaDataAtom)
+  const injectedWidgetMetaData = useInjectedWidgetMetaData()
```

Comment:
nitpick

not comming from this PR, but in principle not sure we want to make all modules aware there's an injected mode and instead is the client of this module informing one time whats the appKey

This way, in injected mode, it will inject the appData, otherwise it will get it from the env.

Otherwise, we leak a lot of CoW Swap logic here. Imagine we want to promote this module to a lib and use for other projects, then the injected mode would be bothering us.

Reply:
Yeah, I also feel it's not the best solution. However, I don't see any better ways